lib/rmt/mirror.rb

```diff
@@ -0,0 +1,86 @@
+module RMT
+end
+
+class RMT::Mirror
+
+  def initialize(mirroring_base_dir, repository_url, local_path, mirror_src = false, logger = nil)
+    @mirroring_base_dir = mirroring_base_dir
+    @repository_url = repository_url
+    @local_path = local_path
+    @mirror_src = mirror_src
+    @logger = logger || Logger.new('/dev/null')
+    @primary_files = []
+    @deltainfo_files = []
+  end
+
+  def mirror
+    mirror_metadata
+    mirror_data
+  end
+
+  protected
+
+  def mirror_metadata
+    temp_dir = Dir.mktmpdir
+
+    @downloader = RMT::Downloader.new(
+      @repository_url,
+      temp_dir.to_s,
+      @logger
+    )
+
+    begin
+      @downloader.download('repodata/repomd.xml.key')
+      @downloader.download('repodata/repomd.xml.asc')
+    rescue RMT::Downloader::Exception # rubocop:disable Lint/HandleExceptions
```

Comment:
The repository might be unsigned, so those files wouldn't exist and the signatures aren't referenced in the metadata.